include/boost/url/bnf/parse.hpp

```diff
@@ -16,58 +16,77 @@
 #include <boost/type_traits/make_void.hpp>
 #include <type_traits>
 
+/** @file
```

Comment:
This looks like a javadoc attached to the file? How can I view this in the documentation output?

Reply:
Yes. This is the official way of doing it in doxygen. If you generate the doxygen output, this will be in the file documentation. Doxygen understands it 100%. 

Whether docca is currently able to understand it is another issue we can handle separately, maybe by moving this content somewhere else. In any case, neither doxygen nor docca would understand what we had before.

Reply:
Okay but the number one rule of writing documentation is that it has to be visible in the output. Otherwise there is nothing to read, and nothing to review.

Reply:
You're 100% correct. That's what I always do too. 

In the general case, the problem here is sometimes we have some conflict between what the correct javadocs/doxygen are and what docca is capable of processing. It might be useful to have an official guideline in this case or for each case. 

In this particular case, for instance, we just had a comment describing how parsing works, which failed to be to be visible in the output either way. I improved that with the proper doxygen command for that, which doxygen parses correctly of course, but docca doesn't allow it to be visible in the output. There is the visible doxygen output:

![image](https://user-images.githubusercontent.com/5369819/147975680-34ff2482-3144-4241-a5a6-d62f4de3bcf4.png)

So in that case, as none of the options would be visible in the output anyway, I inferred that it would be better to have the correct javadocs already instead of making the maintaining wrong javadocs that both docca and doxygen can't parse anyway. At least, in the first alternative, it would be visible output once docca catches up and it's visible for anyone who runs doxygen on the source files. Or we can have an intermediary solution that is replicating this content somewhere else until docca catches up. 

This is not the only place where this happens. In each case, we might decide to use proper documentation until docca catches up or do the opposite and rewrite everything once docca catches up. There are short-term and long-term pros and cons to each approach.

Reply:
No one runs doxygen and produces the HTML output, so these javadocs are unobservable and untestable. If you want to have this information you need to figure out a good way to present it. I would put it in the exposition rather than trying to make it a javadoc.

Reply:
> No one runs doxygen and produces the HTML output, so these javadocs are unobservable and untestable

Yes. So is what we had before? Should it be removed then?

Reply:
>  I would put it in the exposition rather than trying to make it a javadoc.

A good place would be a future `bnf_element` (or something) concept, but I don't even think there's a canonical way to document these yet.

Reply:
> No one runs doxygen and produces the HTML output

Yes. I just showed you the doxygen output because it justifies the rationale for what I discussed late. We do have a problem with different capabilities between doxygen and docca.

Reply:
> We do have a problem with different capabilities between doxygen and docca.

There's no problem. Javadocs in source code must be limited to what docca supports. This has worked for Beast, StaticString, JSON.... and now URL

Reply:
> There's no problem. 

OK. 

> Javadocs in source code must be limited to what docca supports

OK. I get your point. I understand things like not being able to document files or group class members as a problem. But that's OK. I understand the point.

Reply:
This is still not fixed

Reply:
I know. The discussion is happening here: https://github.com/CPPAlliance/url/pull/97#discussion_r777702510